account_invoice_report_grouped_by_picking/views/report_invoice.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="report_invoice_document" inherit_id="account.report_invoice_document" priority="99">
+    <xpath expr="//tbody[@class='invoice_tbody']" position="before">
+        <t t-set="subtotal" t-value="0.0"/>
+    </xpath>
+    <xpath expr="//tr[@t-foreach='o.invoice_line_ids']" position="attributes">
+        <attribute name="t-foreach">o.lines_grouped_by_picking()</attribute>
+        <attribute name="t-as">lines_group</attribute>
+    </xpath>
+    <xpath expr="//td/span[@t-field='l.name']/.." position="before">
+        <t t-set="l" t-value="lines_group['line']"/>
+        <t t-set="picking" t-value="lines_group['picking']"/>
+        <t t-if="picking != last_picking">
+            <tr t-if="last_picking">
+                <td colspan="10" class="text-right">
+                    <strong>Subtotal: </strong>
+                    <strong t-esc="subtotal"
+                            t-options='{"widget": "monetary", "display_currency": "o.currency_id"}'/>
+                </td>
+                <t t-set="subtotal" t-value="0.0"/>
+            </tr>
+            <tr t-if="picking">
+                <td colspan="10">
+                    <strong t-esc="'Order: %s Picking: %s %s' % (picking.sale_id.name, picking.date_done[:10], picking.name)"/>
+                </td>
+            </tr>
+            <t t-set="last_picking" t-value="picking"/>
+        </t>
+    </xpath>
+    <xpath expr="//td/span[@t-field='l.quantity']" position="before">
+        <t t-if="picking">
+            <!-- Compute product qty for that picking and invoice line -->
+            <t t-set="line_picking_qty"
+               t-value="sum(l.mapped('move_line_ids').filtered(lambda x: x.picking_id == picking).mapped('product_uom_qty'))"/>
+            <span t-esc="line_picking_qty"/>
+        </t>
+    </xpath>
+    <xpath expr="//td/span[@t-field='l.quantity']" position="attributes">
+        <attribute name="t-if">not picking</attribute>
+    </xpath>
+    <xpath expr="//td/span[@t-field='l.price_subtotal']" position="before">
+        <t t-if="picking">
+            <!-- Compute subtotal for that picking with discounts -->
+            <t t-set="line_picking_subtotal" t-value="line_picking_qty * (l.price_subtotal / l.quantity)"/>
+            <span t-esc="line_picking_subtotal"
+                  t-options='{"widget": "monetary", "display_currency": "o.currency_id"}'/>
+            <t t-set="subtotal" t-value="(subtotal or 0.0) + line_picking_subtotal"/>
+        </t>
+    </xpath>
+    <xpath expr="//td/span[@t-field='l.price_subtotal']" position="attributes">
+        <attribute name="t-if">not picking</attribute>
+    </xpath>
+    <xpath expr="//tbody[@class='invoice_tbody']" position="inside">
```

Comment:
This block has to be removed:

* The position is very relaxed, being placed when multiple inheritance randomly.
* You shouldn't see latest picking subtotal this way.